From 91924101a5ed0862ef78d60b732c13b0581c237c Mon Sep 17 00:00:00 2001
From: Fancy Fang <chen.fang@freescale.com>
Date: Thu, 5 Mar 2015 14:03:34 +0800
Subject: [PATCH] MLK-10373-6 ARM: clk_imx7d: remove the lcdif pix clk enable
 by default

Remove the lcdif pixel clock enable by default in
the CCM driver. This clock can be handled in the
mxsfb driver correctly later.

Signed-off-by: Fancy Fang <chen.fang@freescale.com>
---
 arch/arm/mach-imx/clk-imx7d.c | 6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)

diff --git a/arch/arm/mach-imx/clk-imx7d.c b/arch/arm/mach-imx/clk-imx7d.c
index 01cdfb4..95239e08 100644
--- a/arch/arm/mach-imx/clk-imx7d.c
+++ b/arch/arm/mach-imx/clk-imx7d.c
@@ -898,8 +898,10 @@ static void __init imx7d_clocks_init(struct device_node *ccm_node)
 	/* set parent of EPDC pixel clock */
 	imx_clk_set_parent(clks[IMX7D_EPDC_PIXEL_ROOT_SRC], clks[IMX7D_PLL_SYS_MAIN_CLK]);
 
-	for (i = 0; i < IMX7D_END_CLK; i++)
-		clk_prepare_enable(clks[i]);
+	for (i = 0; i < IMX7D_END_CLK; i++) {
+		if (likely(i != IMX7D_LCDIF_PIXEL_ROOT_CLK))
+			clk_prepare_enable(clks[i]);
+	}
 
 	mxc_timer_init_dt(of_find_compatible_node(NULL, NULL, "fsl,imx7d-gpt"));
 }
-- 
1.8.0

